tests/test_rl_graders.py

```diff
@@ -0,0 +1,35 @@
+from typing import Iterator, List
+import unittest
+
+from rlbot.training.training import Pass, Fail, FailDueToExerciseException
+
+from rlbottraining.exercise_runner import run_playlist
+from rlbottraining.history.exercise_result import ExerciseResult
+
+class rl_grader_tester(unittest.TestCase):
+    '''
+    This tests the grader that simulates rocket league environments, like the shooter training pack
+    '''
+
+    def assertGrades(self, result_iter: Iterator[ExerciseResult], want_grades: List[str]):
+        got_grades = []
+        for result in result_iter:
+            if isinstance(result.grade, FailDueToExerciseException):
+                self.fail(str(result.grade))
+                break
+            got_grades.append(result.grade.__class__.__name__)
+        self.assertEqual(got_grades, want_grades)
+
+    def test_rl_graders(self):
+        from tests.test_exercises.rl_grader_exercises import make_default_playlist
+        self.assertGrades(
+            run_playlist(make_default_playlist()),
```

Comment:
I think it's easier to understand the test case if the exercises are constructed in this function.
Main reason is to make it easy to see which grade corresponds with which exercise.
Note: the reason the other tests work differently is because the make_default_playlist() has other uses as well, whereas here it does not.

Reply:
Ok, i will keep working on a separate file until I have all the tests done, then I will move it here.